src/Microsoft.AspNetCore.OData/Results/NotFoundODataResult.cs

```diff
@@ -0,0 +1,74 @@
+//-----------------------------------------------------------------------------
+// <copyright file="NotFoundODataResult.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.OData;
+using ErrorUtils = Microsoft.AspNetCore.OData.Error;
+
+namespace Microsoft.AspNetCore.OData.Results
+{
+    /// <summary>
+    /// Represents a result that when executed will produce a Not Found (404) response.
+    /// </summary>
+    /// <remarks>This result creates an <see cref="ODataError"/> with status code: 404.</remarks>
+    public class NotFoundODataResult : NotFoundResult, IODataErrorResult
+    {
+        private const string errorCode = "404";
+
+        /// <summary>
+        /// OData Error.
+        /// </summary>
+        public ODataError Error { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the class.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        public NotFoundODataResult(string message)
+        {
+            if (string.IsNullOrEmpty(message))
+            {
+                throw ErrorUtils.ArgumentNullOrEmpty(nameof(message));
+            }
+
+            Error = new ODataError
+            {
+                Message = message,
+                ErrorCode = errorCode
+            };
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the class.
+        /// </summary>
+        /// <param name="odataError">OData Error.</param>
```

Comment:
```suggestion
        /// <param name="odataError">OData error.</param>
```
Or: `<param name="odataError">An <see cref="ODataError"/> object.</param>`
Or: `<param name="odataError">An <see cref="ODataError"/> instance.</param>`